ksql-functional-tests/src/test/resources/query-validation-tests/elements.json

```diff
@@ -574,10 +602,30 @@
         "CREATE STREAM OUTPUT AS SELECT ROWKEY FROM INPUT;"
       ],
       "expectedException": {
-        "type": "io.confluent.ksql.util.KsqlException",
+        "type": "io.confluent.ksql.util.KsqlStatementException",
         "message": "Reserved column name in select: `ROWKEY`. Please remove or alias the column."
       }
     },
+    {
+      "name": "non-join should reject KEY column name in projection",
+      "comments": [
+        "changes to the ksql query semantics are required to allow this.",
+        "At the moment, the key schema passes through the select un-changed.",
+        "which means the key column in the projection is added to the value schema",
+        "but the name of the column clashes with the key column, resulting in an error"
+      ],
+      "statements": [
+        "CREATE STREAM INPUT (K INT KEY, F0 INT) WITH (kafka_topic='input', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT K FROM INPUT;"
+      ],
+      "properties": {
+        "ksql.any.key.name.enabled": true
+      },
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Value column name(s) `K` clashes with key column name(s)."
```

Comment:
I think as part of this error message we should explain the semantics (that `K` will be available in the key schema)

Reply:
Not sure I follow.   What is it you're looking for?

It's already saying the value column name clashes with the key column name...

I don't think it's always safe to say the data held in `K` will be available via the key column of the same name: it could be completely different data, just aliased to `K`.

Reply:
Concerns about error message hopefully addressed in https://github.com/confluentinc/ksql/pull/4739